### PR TITLE
[BEAM-5837] Add dashboard for monitoring input data staleness

### PR DESCRIPTION
**Pull request by swegner**
This will make it easier to monitor whether we have fresh input data in
in our dashboards. We can set up notifications or probing against the
dahsboard alerts.

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_GradleBuild/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_GradleBuild/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Gradle/lastCompletedBuild/) [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark_Gradle/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | --- | ---

### CODE REVIEW

**Comment by swegner** [comment]
R: @Ardagan

**Comment by swegner** [comment]
Live demo: http://104.154.241.245/d/data-freshness/source-data-freshness?edit&orgId=1

**Comment by swegner** [comment]
The alert needs to be changed to 8 hours instead of mins

File: .test-infra/metrics/src/test/groovy/ProberTests.groovy [hunk 1]
```diff
@@ -36,7 +38,17 @@ class ProberTests {
     def dashboardNames = allDashboards.title
     // Validate at least one expected dashboard exists
     assert dashboardNames.contains('Post-commit Tests') : 'Expected dashboard does not exist'
-    assert dashboardNames.size >= 3 : "Number of dashboards is less than expected ${dashboardNames}"
+    assert dashboardNames.size >= 4 : "Number of dashboards is less than expected ${dashboardNames}"
```
Comment (Ardagan) [inline .test-infra/metrics/src/test/groovy/ProberTests.groovy]: This test is too strict. Will fire false alarms on each code deployment or change. 
```suggestion
    assert dashboardNames.size > 0 : "No dashboards found. Check Grafana dashboard initialization script."
```

**Comment by swegner** [comment]
Feedback addressed; @Ardagan PTAL

**Pull request merged**